src/ext/legit/legit-commit.lisp

```diff
@@ -0,0 +1,125 @@
+     :keymap *legit-commit-mode-keymap*)
+  ;; no syntax highlihgt in fact.
+  (setf (variable-value 'enable-syntax-highlight) t))
+
+;; Validate, abort.
+(define-key *legit-commit-mode-keymap* "C-c C-c" 'commit-continue)
+(define-key *legit-commit-mode-keymap* "C-Return" 'commit-continue)
+(define-key *legit-commit-mode-keymap* "M-q" 'commit-abort)
+(define-key *legit-commit-mode-keymap* "C-c C-k" 'commit-abort)
+
+;; Navigation.
+;; find and display the previous commit messages.
+;; (define-key *legit-commit-mode-keymap* "C-n" 'next-commit)
+;; (define-key *legit-commit-mode-keymap* "C-p" 'previous-commit)
+
+;; Help.
+(define-key *legit-commit-mode-keymap* "C-x ?" 'commit-help)
+
+(defun commit ()
+  (let ((buffer (make-buffer "*legit-commit*")))
+    (setf (buffer-directory buffer) (uiop:getcwd))
+    (setf (buffer-read-only-p buffer) nil)
+    (erase-buffer buffer)
+    (move-to-line (buffer-point buffer) 1)
+    (insert-string (buffer-point buffer) "write commit message:")
+    (change-buffer-mode buffer 'legit-commit-mode)
+    ;; (setf (buffer-read-only-p buffer) t)
+    (move-to-line (buffer-point buffer) 1)))
+
+(defun clean-commit-message (text)
+  "Remove lines starting with a #."
+  ;; We should collect meaningful data too, like a signature.
+  (loop for line in (str:lines text)
+        unless (str:starts-with-p "#" line)
+          collect line into result
+        finally (return (str:unlines result))))
+
+;; void message:
+#+(or)
+(assert (str:blankp (clean-commit-message
+"# Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+")))
+
+;; a message on the first line:
+#+(or)
+(assert (equal "test message" (clean-commit-message
+"test message
+# Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+")))
+
+;; a few lines:
+#+(or)
+(assert (equal "one
+
+two
+" (clean-commit-message
+"one
+
+two
+
+# Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+")))
+
+(defun commit-continue ()
+  (let* ((message (buffer-text (make-buffer "*legit-commit*")))
+         (cleaned-message (clean-commit-message message)))
+    (cond
+      ((str:blankp cleaned-message)
+       (message "No commit message, do nothing."))
+      (t
+       (with-current-project ()
+         (run-function (lambda ()
+                         (lem/porcelain::commit cleaned-message))
+                       :message "commited")
+         (kill-buffer "*legit-commit*")
+         ;; come back on the status on  the left:
+         (lem-core/commands/window:previous-window)
+         ;; and refresh.
+         (legit-status))))))
+
+(define-command commit-abort () ()
+  (kill-buffer "*legit-commit*")
+  (lem-core/commands/window:previous-window))
+
+(define-command commit-abort-yes-or-no () ()
+  ;; TODO: prompt for confirmation.
```

Comment:
Can we use prompt-for-y-or-n-p?

Reply:
\+1 good catch thanks. Do we have a prompt where Y is the default choice? (so we can only press Enter, not "y")